crates/rpc/rpc-eth-types/src/cache/mod.rs

```diff
@@ -178,6 +178,20 @@ impl EthStateCache {
         Ok(block.zip(receipts))
     }
 
+    /// Fetches receipts and optionally fetches the block if available in the cache.
+    /// If the block is not in the cache, it only returns the receipts.
+    pub async fn get_receipts_and_maybe_block(
+        &self,
+        block_hash: B256,
+    ) -> ProviderResult<Option<(Arc<Vec<Receipt>>, Option<Arc<SealedBlockWithSenders>>)>> {
+        let block = self.get_sealed_block_with_senders(block_hash);
+        let receipts = self.get_receipts(block_hash);
+
+        let (block, receipts) = futures::try_join!(block, receipts)?;
+
+        Ok(receipts.map(|r| (r, block)))
```

Comment:
this isn't very useful because this is more or less the same thing as 

https://github.com/paradigmxyz/reth/blob/b8147708ad6ad997c14c2a11b4ed5196bdd38e4b/crates/rpc/rpc-eth-types/src/cache/mod.rs#L169-L169

Reply:
Agreed, that's why I was unsure of this change, but I don't fully understand what else can I do because:

Currently, get_sealed_block_with_senders checks the block cache, and if the block isn’t cached, it queues a request to the provider via CacheAction::GetBlockWithSenders. This request is async, and the result (either the block or an error) is sent back later. However, it doesn’t return anything immediately if the block isn’t cached. 

So the new function (get_receipts_and_maybe_block) always tries to retrieve the receipts, and checks the block (get_sealed_block_with_senders), by using map instead of zip we make sure if block is None we still retrieve the receipts. I think this is the only relevant change needed... We can relax the condition even more by returning None if get_sealed_block_with_senders returned an error. Is that what you mean?

Or do I rather should introduce a new cache action that just checks the block in the cache but does not trigger a request from the provider if it does not find it?